api/notd/collection_processor.py

```diff
@@ -44,15 +44,15 @@ async def retrieve_collection(self, address: str) -> RetrievedCollection:
             except ResponseException as exception:
                 if exception.statusCode == 404:
                     raise NotFoundException()
-                if exception.statusCode != 429 or retryCount >= 3:
+                if exception.statusCode == 429:
```

Comment:
ah this definitely looks wrong but its hard to even articulate why. I'll write out some test cases please test and ensure it does all these:

if 404 raised, bail out straight away and raise CollectionDoesNotExist
if >=500 raised, bail out straight away and raise CollectionDoesNotExist
if retry count >= 3 bail out straight away with whatever exception should happen

Reply:
in fact the retry logic should just be that on 429 ONLY it retries. if its a 404 or 500 it should raise CollectionDoesNotExist and for any other exception it should raise whatever exception was caught.